src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java

```diff
@@ -436,8 +440,11 @@ public void testAllCheckstyleModulesHaveXdocDocumentation() throws Exception {
     @Test
     public void testAllCheckstyleModulesInCheckstyleConfig() throws Exception {
         final Set<String> configChecks = CheckUtil.getConfigCheckStyleModules();
+        final Set<String> moduleNames = CheckUtil.getSimpleNames(CheckUtil.getCheckstyleModules());
+        //Issue: https://github.com/checkstyle/checkstyle/issues/4421
+        moduleNames.remove("SuppressionXpathFilter");
```

Comment:
Just to confirm, so we are doing this in the same issue not a different one?

Reply:
Should I specify current issue or issue when it will be resolved?

Reply:
@rnveach 
I do not understand your question. What should be done in the different issue? SuppressionXpathFilter is not ready to be used in our config, so it is removed from validation. We will activate it when the model will be ready and available to be configured and used in our code.

Reply:
@timurt , please make a comment like "till ...<link to issue>.." to make it clear when it should be resolved.

All modules HAVE to be used in our config, if there some problems it is always till some issue is resolved/implementated , but there are always a reason.

Reply:
> We will activate it when the model is ready and available to be configured and used in our code.

My point is that we can't close the current issue then until we put the filter in our config as the comment says until this issue.
If this PR is merged, are we going to start working on this next? If so, then that is fine with me.
If we have to do other things before we can put this in our  config, then I think it should be a separate issue so we can close the current one after this PR.
We don't usually like to leave issues with partial work done open for long.